Source/WTF/Scripts/generate-platform-args

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import subprocess
+import os
+import shlex
+import shutil
+import re
+
+# Preprocesses wtf/Platform.h with clang, extracts macro definitions, and saves
+# the platform flags to a Swift-style response file containing -D arguments.
+
+framework_search_paths = shlex.split(
+    '{BUILT_PRODUCTS_DIR} {FRAMEWORK_SEARCH_PATHS}'.format_map(os.environ))
+header_search_paths = shlex.split(
+    '{BUILT_PRODUCTS_DIR} {HEADER_SEARCH_PATHS}'.format_map(os.environ))
+system_framework_search_paths = shlex.split(
+    '{SYSTEM_FRAMEWORK_SEARCH_PATHS}'.format_map(os.environ))
+system_header_search_paths = shlex.split(
+    '{SYSTEM_HEADER_SEARCH_PATHS}'.format_map(os.environ))
+preprocessor_definitions = shlex.split(
+    'RELEASE_WITHOUT_OPTIMIZATIONS {GCC_PREPROCESSOR_DEFINITIONS}'.format_map(os.environ))
```

Comment:
I'm realizing that this is all intricate enough that it would benefit from regression tests. Not saying that you need to add hem before landing, but concerned that this script will be hard to maintain.

Just as an example, I cannot tell why `RELEASE_WITHOUT_OPTIMIZATIONS` is here.

Reply:
RELEASE_WITHOUT_OPTIMIZATIONS is here because I copied the original logic from DerivedSources.make in WebCore: https://github.com/WebKit/WebKit/blob/01b4cde270584956e84a2caf200480691f4884ce/Source/WebCore/DerivedSources.make#L44-L48

My thought (hope!) is that Swift code will depend on this logic enough that regressions here will cause failures down the line, though specific details like this won't be caught.